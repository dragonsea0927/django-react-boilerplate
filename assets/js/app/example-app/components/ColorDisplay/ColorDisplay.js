import React from 'react';
import PropTypes from 'prop-types';


const ColorDisplay = (props) => {
  const { color } = props;

  const style = {
    color,
    fontWeight: 'bold',
  };

  return (
    <span style={style}>
      {color}
    </span>
  );
};

ColorDisplay.defaultProps = {
  color: 'black',
};

ColorDisplay.propTypes = {
  color: PropTypes.string,
};

export default ColorDisplay;
