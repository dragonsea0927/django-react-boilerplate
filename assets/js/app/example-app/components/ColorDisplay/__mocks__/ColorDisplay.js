import React from 'react';


const mock = props => (
  <span>ColorDisplay {JSON.stringify(props)}</span>
);

export default mock;
