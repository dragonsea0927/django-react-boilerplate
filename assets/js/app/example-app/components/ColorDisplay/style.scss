%text-color {
  font-weight: bold;
  margin: 0px 5px;
};

.color-black {
  @extend %text-color;
  color: black;
}

.color-green {
  @extend %text-color;
  color: green;
}

.color-red {
  @extend %text-color;
  color: red;
}

.color-blue {
  @extend %text-color;
  color: blue;
}

.color-purple {
  @extend %text-color;
  color: purple;
}
