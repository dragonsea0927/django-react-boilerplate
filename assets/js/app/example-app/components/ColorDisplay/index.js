import ColorDisplay from './ColorDisplay';

export default ColorDisplay;
