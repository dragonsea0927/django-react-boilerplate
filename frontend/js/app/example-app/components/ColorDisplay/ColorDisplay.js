import PropTypes from 'prop-types';
import React from 'react';

import './style.scss';

const ColorDisplay = (props) => {
  const { color } = props;

  return <span className={`color-${color}`}>{color}</span>;
};

ColorDisplay.defaultProps = {
  color: 'black',
};

ColorDisplay.propTypes = {
  color: PropTypes.string,
};

export default ColorDisplay;
