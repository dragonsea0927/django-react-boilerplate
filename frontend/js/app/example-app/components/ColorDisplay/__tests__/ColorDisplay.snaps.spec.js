import { shallow } from 'enzyme';
import React from 'react';

import ColorDisplay from '../ColorDisplay';

describe('ColorDisplay', () => {
  test('purple', () => {
    const wrapper = shallow(<ColorDisplay color="purple" />);

    expect(wrapper).toMatchSnapshot();
  });

  test('no color (should default to black)', () => {
    const wrapper = shallow(<ColorDisplay />);

    expect(wrapper).toMatchSnapshot();
  });

  test('unknown color', () => {
    const wrapper = shallow(<ColorDisplay color="caterpillar" />);

    expect(wrapper).toMatchSnapshot();
  });
});
