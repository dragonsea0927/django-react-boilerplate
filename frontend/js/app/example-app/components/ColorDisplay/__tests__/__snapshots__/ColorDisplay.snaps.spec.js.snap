// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorDisplay no color (should default to black) 1`] = `
<span
  className="color-black"
>
  black
</span>
`;

exports[`ColorDisplay purple 1`] = `
<span
  className="color-purple"
>
  purple
</span>
`;

exports[`ColorDisplay unknown color 1`] = `
<span
  className="color-caterpillar"
>
  caterpillar
</span>
`;
